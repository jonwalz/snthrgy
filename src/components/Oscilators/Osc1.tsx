import {useAtom} from "jotai";
import React, {useState} from "react";
import {osc1Range} from "../../store/oscillators";
import {Dial} from "../Dial/Dial";
import {
  DialContainer,
  SectionTitle,
  SubsectionContainer,
} from "../sharedStyles";

export default function Osc1() {
  // Move states into atoms
  const [freq1Value, setFreq1Value] = useAtom(osc1Range);
  const [osc2Value, setOsc2Value] = useState(0);
  const [osc3Value, setOsc3Value] = useState(0);

  return (
    <SubsectionContainer>
      <SectionTitle>Osc 1</SectionTitle>
      <DialContainer>
        <Dial
          value={osc2Value}
          diameter={50}
          min={0}
          max={120}
          step={1}
          onValueChange={setOsc2Value}
        >
          Wave
        </Dial>
        <Dial
          value={freq1Value}
          diameter={180}
          min={0}
          max={2000}
          step={1}
          onValueChange={setFreq1Value}
        >
          <div>Frequency</div>
          <div>{freq1Value}</div>
        </Dial>
        <Dial
          value={osc3Value}
          diameter={50}
          min={0}
          max={120}
          step={1}
          onValueChange={setOsc3Value}
        >
          Pan
        </Dial>
      </DialContainer>
    </SubsectionContainer>
  );
}
