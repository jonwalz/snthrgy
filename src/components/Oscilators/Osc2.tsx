import React, {useState} from "react";
import {Dial} from "../Dial/Dial";
import {DialContainer, OscContainer, OscTitle} from "./styles";

export default function Osc2() {
  const [osc1Value, setOsc1Value] = useState(0);
  const [osc2Value, setOsc2Value] = useState(0);
  const [osc3Value, setOsc3Value] = useState(0);

  return (
    <OscContainer>
      <OscTitle>Osc 2</OscTitle>
      <DialContainer>
        <Dial
          value={osc1Value}
          diameter={50}
          min={0}
          max={120}
          step={1}
          onValueChange={setOsc1Value}
        >
          Wave
        </Dial>
        <Dial
          value={osc2Value}
          diameter={50}
          min={0}
          max={120}
          step={1}
          onValueChange={setOsc2Value}
        >
          Tune
        </Dial>
        <Dial
          value={osc3Value}
          diameter={50}
          min={0}
          max={120}
          step={1}
          onValueChange={setOsc3Value}
        >
          Pan
        </Dial>
      </DialContainer>
    </OscContainer>
  );
}
