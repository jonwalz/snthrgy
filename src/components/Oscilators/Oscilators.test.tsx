import React from "react";
import {screen, render} from "@testing-library/react";
import Osc1 from "./Osc1";
import Osc2 from "./Osc2";

describe("Oscilators", () => {
  test("Osc1 should render correctly", () => {
    const {asFragment} = render(<Osc1 />);

    screen.getByText("Osc 1");
    screen.getByText("Wave");
    screen.getByText("Tune");
    screen.getByText("Pan");

    expect(asFragment()).toMatchSnapshot();
  });

  test("Osc2 should render correctly", () => {
    const {asFragment} = render(<Osc2 />);

    screen.getByText("Osc 2");
    screen.getByText("Wave");
    screen.getByText("Tune");
    screen.getByText("Pan");

    expect(asFragment()).toMatchSnapshot();
  });
});
