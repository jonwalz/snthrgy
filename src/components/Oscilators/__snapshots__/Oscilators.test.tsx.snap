// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oscilators Osc1 should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz dbtrjS"
  >
    <h4
      class="sc-hKgILt ZTQiC"
    >
      Osc 1
    </h4>
    <div
      class="sc-dlfnbm hPNVhb"
    >
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Wave
      </div>
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Tune
      </div>
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Pan
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Oscilators Osc2 should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz dbtrjS"
  >
    <h4
      class="sc-hKgILt ZTQiC"
    >
      Osc 2
    </h4>
    <div
      class="sc-dlfnbm hPNVhb"
    >
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Wave
      </div>
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Tune
      </div>
      <div
        class="sc-bdfBwQ jsEWwN"
      >
        <div
          aria-valuemax="120"
          aria-valuemin="0"
          aria-valuenow="0"
          style="width: 50px; height: 50px; border-radius: 25px; position: relative; outline: none; box-sizing: border-box; overflow: hidden; cursor: pointer;"
          tabindex="0"
        >
          <svg
            style="transform: rotate(0deg);"
            transform="rotate(0)"
          >
            dial.svg
          </svg>
        </div>
        Pan
      </div>
    </div>
  </div>
</DocumentFragment>
`;
